### ISSUE TITLE
Ryslog 8.2212.0 fails to start in docker environment

### ISSUE THREAD

**Issue by juanbrein** [rsyslog/rsyslog#5096]
### Expected behavior

Run rsyslog in docker

### Actual behavior

Errors with:

rsyslogd 8.2302.0: running as pid 1, enabling container-specific defaults, press ctl-c to terminate rsyslog
rsyslog internal message (3,-2455): could not transfer  the  specified  internal posix  capabilities settings to the kernel, capng_apply=-5
 [v8.2302.0 try https://www.rsyslog.com/e/2455 ]

### Steps to reproduce the behavior

Run rsyslog 8.2212.0 within a docker container

### Environment
- rsyslog version: 8.2212.0
- platform: docker/alpine
- for configuration questions/issues, include rsyslog.conf and included config files

**Comment by juanbrein** [comment]
Something changed in the latest rsync release that is now breaking on docker environments. 

The easy fix is to run the container using `privileged: true`. This is ok for some use cases... but for some others imply a security compromise. 

Is there a way I can tweak rsyslog to not request that capability at run time?

**Comment by rgerhards** [comment]
where did you install/build from? It looks like libpcap-ng is enabled, which IMHO makes little sense here.

@Cropi anything we can work around if enabled?

**Comment by juanbrein** [comment]
alpine:edge docker image

**Comment by rgerhards** [comment]
> should I raise this with the alpine teams?

yes, this would make a lot of sense IMHO. I do not see any reason to compile rsyslog for docker with libpcap_ng enabled. In any case, disabling it is ast least a work-around.

**Comment by juanbrein** [comment]
Is there a way to disable it from the command line/config file/set environment variable at run time?

**Comment by rgerhards** [comment]
unfortunately no

**Comment by juanbrein** [comment]
This now has been resolved. Thanks!

**Comment by dermotbradley** [comment]
> I do not see any reason to compile rsyslog for docker with libpcap_ng enabled.

This is not "rsyslog for docker", this is the Alpine distro package for rsyslog - Alpine is used for physical and virtual machines as well as for containers.

The libcap functionality makes sense for when rsyslog is used on physical and virtual machine to reduce attack surface.

Perhaps rsyslog could add a new configuration setting to control whether or not to treat the ```capng_update``` and ```capng_apply``` failures as merely warnings? Then someone using the binary in a container could switch the setting to only warn.

**Comment by vizovitin** [comment]
This now fails in `almalinux:9` (and likely other RedHat el9 based images).
